app-ui-catalog/src/main/java/app/k9mail/ui/catalog/ui/atom/items/IconItems.kt

```diff
@@ -24,19 +24,23 @@ fun LazyGridScope.iconItems() {
 }
 
 private inline fun <reified T> LazyGridScope.getIconsFor(icons: T) {
-    for (field in T::class.java.declaredFields) {
-        if (field.name in exclusions) continue
-        item {
-            field.isAccessible = true
-            IconItem(
-                name = field.name.replaceFirstChar { it.uppercase() },
-                imageVector = field.get(icons) as ImageVector,
-            )
+    for (method in T::class.java.methods) {
```

Comment:
If you include kotlin-reflect this can be written as:

```kotlin
private inline fun <reified T : Any> LazyGridScope.getIconsFor(icons: T) {
    for (property in T::class.declaredMemberProperties) {
        item {
            IconItem(
                name = property.name.replaceFirstChar { it.uppercase() },
                imageVector = property.get(icons) as ImageVector,
            )
        }
    }
}
```

Reply:
That would be simpler indeed, but it wanted to avoid introducing Kotlin Reflect.